src/status_im/ui/screens/routing/core.cljs

```diff
@@ -118,6 +118,13 @@
       (utils/update-if-present :initialRouteName name)
       (utils/update-if-present :mode name)))
 
+(defn new-get-state-for-action [default-get-state-for-action]
+  (fn [action state]
+    (let [event (get back-actions/back-actions @view-id)]
+      (when (and (= (.-type action) (.-BACK navigation/navigation-actions)) event (not= :default event))
+        (re-frame/dispatch [event]))
```

Comment:
This dispatch is never invoked currently. So this means that `create.core/intro-step-back` is never invoked on Android hardware back or iOS swipe, and `recover.core/cancel-pressed` is never invoked for actions included above + clicking on the Back arrow. So wizard state is not updated. This causes e.g. these issues:
 1) If one presses Android's hardware back on Onboarding's `Select account` screen, and then generates keys again, they are redirected to `Select key storage` screen bypassing `Select account` screen
2) On Basic recovery's `Select storage` screen clicking on Cancel is not going to show any confirmation dialog.

Reply:
i didn't whitelist onboarding screens, so they will work fine because you can't use hardware buttons on them

Reply:
don't want to mess with wizard in this PR, but we could try to get rid of it for sure